components/match2/wikis/criticalops/match_group_input_custom.lua

```diff
@@ -0,0 +1,683 @@
+
+			-- Retrieve icon for team
+			if opponent.type == Opponent.team then
+				opponent.icon, opponent.icondark = opponentFunctions.getIcon(opponent.template)
+			end
+
+			-- apply status
+			if TypeUtil.isNumeric(opponent.score) then
+				opponent.status = 'S'
+				isScoreSet = true
+			elseif Table.includes(ALLOWED_STATUSES, opponent.score) then
+				opponent.status = opponent.score
+				opponent.score = -1
+			end
+			opponents[opponentIndex] = opponent
+
+			-- get players from vars for teams
+			if opponent.type == Opponent.team and not Logic.isEmpty(opponent.name) then
+				match = matchFunctions.getPlayers(match, opponentIndex, opponent.name)
+			end
+		end
+	end
+
+	-- Handle tournament status for unfinished matches
+	if (not Logic.readBool(match.finished)) and Logic.isNotEmpty(match.status) then
+		match.finished = match.status
+	end
+
+	if Table.includes(NP_MATCH_STATUS, match.finished) then
+		match.resulttype = 'np'
+		match.status = match.finished
+		match.finished = false
+		match.dateexact = false
+	else
+		-- see if match should actually be finished if score is set
+		if isScoreSet and not Logic.readBool(match.finished) and match.hasDate then
+			local currentUnixTime = os.time(os.date('!*t'))
+			local lang = mw.getContentLanguage()
+			local matchUnixTime = tonumber(lang:formatDate('U', match.date))
+			local threshold = match.dateexact and 30800 or 86400
+			if matchUnixTime + threshold < currentUnixTime then
+				match.finished = true
+			end
+		end
+
+		if Logic.readBool(match.finished) then
+			match, opponents = CustomMatchGroupInput.getResultTypeAndWinner(match, opponents)
+		end
+	end
+
+	-- Update all opponents with new values
+	for opponentIndex, opponent in pairs(opponents) do
+		match['opponent' .. opponentIndex] = opponent
+	end
+	return match
+end
+
+-- Get Playerdata from Vars (get's set in TeamCards)
+function matchFunctions.getPlayers(match, opponentIndex, teamName)
+	-- match._storePlayers will break after the first empty player. let's make sure we don't leave any gaps.
+	local count = 1
+	for playerIndex = 1, MAX_NUM_PLAYERS do
+		-- parse player
+		local player = match['opponent' .. opponentIndex .. '_p' .. playerIndex] or {}
+		player = Json.parseIfString(player)
+		local playerPrefix = teamName .. '_p' .. playerIndex
+		player.name = player.name or Variables.varDefault(playerPrefix)
+		player.flag = player.flag or Variables.varDefault(playerPrefix .. 'flag')
+		player.displayname = player.displayname or Variables.varDefault(playerPrefix .. 'dn')
+		if not Table.isEmpty(player) then
+			match['opponent' .. opponentIndex .. '_p' .. count] = player
+			count = count + 1
+		end
+	end
+	return match
+end
+
+--
+-- map related functions
+--
+
+-- Check if a map should be discarded due to being redundant
+-- DUMMY_MAP_NAME needs the match the default value in Template:Map
+function mapFunctions.discardMap(map)
+	if map.map == DUMMY_MAP_NAME then
+		return true
+	else
+		return false
+	end
```

Comment:
```suggestion
	return map.map == DUMMY_MAP_NAME
```
maybe even ditch the function and just do the plain check instead of calling it (since the function is only used at 1 location)